docs/gitbook/tutorials/osm-progressive-delivery.md

```diff
@@ -0,0 +1,360 @@
+# Open Service Mesh Canary Deployments
+
+This guide shows you how to use Open Service Mesh (OSM) and Flagger to automate canary deployments.
+
+![Flagger OSM Traffic Split](https://raw.githubusercontent.com/fluxcd/flagger/main/docs/diagrams/flagger-osm-traffic-split.png)
+
+## Prerequisites
+
+Flagger requires a Kubernetes cluster **v1.16** or newer and Open Service Mesh **0.9.1** or newer.
+
+Install Open Service Mesh with Prometheus and permissive traffic policy enabled.
+
+```bash
+osm install \
+--set=OpenServiceMesh.deployPrometheus=true \
+--set=OpenServiceMesh.enablePermissiveTrafficPolicy=true
+```
+
+Install Flagger in the `osm-system` namespace.
+
+```bash
+kubectl apply -k github.com/fluxcd/flagger//kustomize/osm
+```
+
+## Bootstrap
+
+Flagger takes a Kubernetes deployment and optionally a horizontal pod autoscaler (HPA),
+then creates a series of objects (Kubernetes deployments, ClusterIP services and SMI traffic split).
+These objects expose the application inside the mesh and drive the canary analysis and promotion.
+
+Create a `test` namespace and enable osm namespace monitoring and metrics scraping for the namespace.
+
+```bash
+kubectl create namespace test
+osm namespace add test
+osm metrics enable --namespace test
+```
+
+Install the load testing service to generate traffic during the canary analysis:
+
+```bash
+kubectl apply -k https://github.com/fluxcd/flagger//kustomize/tester?ref=main
+```
+
+Enable inbound port exclusion on the Flagger loadtester so that the Flagger controller (in the `osm-system` namespace) can reach the Flagger loadtester (in the `test` namespace).
+
+```bash
+kubectl patch \
+deployment flagger-loadtester \
+-n test \
+-p '{"spec": {"template": {"metadata": {"annotations": {"openservicemesh.io/inbound-port-exclusion-list": "80, 8080"}}}}}' \
+--type=merge
+```
+
+Restart the `flagger-loadtester` deployment to reflect the changes.
+
+```bash
+kubectl rollout restart deploy/flagger-loadtester -n test
+```
+
+Create a `podinfo` deployment and a horizontal pod autoscaler:
+
+```bash
+kubectl apply -k https://github.com/fluxcd/flagger//kustomize/podinfo?ref=main
+```
+
+Create a canary custom resource for the `podinfo` deployment.
+The following `podinfo` canary custom resource instructs Flagger to:
+1. Monitor any changes to the `podinfo` deployment created earlier,
+2. Detect `podinfo` deployment revision changes, and
+3. Start a Flagger canary analysis, rollout, and promotion if there were deployment revision changes.
+
+```yaml
+apiVersion: flagger.app/v1beta1
+kind: Canary
+metadata:
+  name: podinfo
+  namespace: test
+spec:
+  # deployment reference

```

Comment:
Let’s add here `provider: osm`, even if it’s not required, it allows people to mix ingress controllers with OSM.

Reply:
Added